Sources/MockoloFramework/Utils/TypeParser.swift

```diff
@@ -542,6 +542,30 @@ public final class Type {
         let typeStr = "((\(displayableParamStr)) \(suffixStr)-> \(displayableReturnType))?"
         return Type(typeStr, cast: returnTypeCast)
     }
+    
+    static func toArgumentsHistoryType(with params: [Type], typeParams: [String]) -> Type {
+        // Expected only history capturable types.
+        let displayableParamTypes = params.compactMap { (subtype: Type) -> String? in
+            var processedType = subtype.processTypeParams(with: typeParams)
+            
+            if subtype.isInOut {
+                processedType = String(processedType.dropFirst((String.inout + " ").count))
+            }
+            if subtype.isIUO {
+                processedType = String(processedType.dropLast("!".count))
+            }
+            
```

Comment:
Can you add handling for generic type params?  See how they are handled in the Closure model (functionHandler).

Reply:
> Can you add handling for generic type params and overloaded funcs with the same name? See how they are handled in the Closure model (function handler).

Handled in df1a3bd and added test cases was passed ✅ 

> Re: overloaded funcs, we can, for now, allow args of all overloaded funcs to be saved if enabled via the annotation 'history' argument. The name of the argumenthistorymodel will need to be unique in this case -- you can follow the naming of the function handler closure model to be consistent.

This means that users must annotate like as following?
```
/// @mockable(history: fooFuncValString = true)
protocol Foo {
    func fooFunc(val: Int)
    func fooFunc(val: String)  // To enable history only for it.
}
```

It is seems confusing to me 😢 
In this case, I think that capturing history of all of fooFuncs isn't bad.

Reply:
Yeah, to simplify, we can allow all overloaded funcs to have args history for now.